ibis/backends/pandas/execution/generic.py

```diff
@@ -112,7 +112,7 @@ def execute_cast_series_array(op, data, type, **kwargs):
             '(e.g., number, string, or timestamp)'
         )
     return data.map(
```

Comment:
Can you micro if this is faster/slower?

```
pd.Series(np.array(data.to_numpy().tolist()).astype(target_dtype).tolist())
```

Reply:
The idea is that to see if putting it into a 2d array makes things faster/slower

Reply:
That makes sense as an alternative to me, however it doesn't seem to work if the arrays in the Series don't all have the same length. (`np.array(data.to_numpy().tolist())` then results in nested `np.arrays` which doesn't work with `.astype(...)`)

Reply:
LGTM